portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,183 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
+/*
+	*Dynamically fetch content from github and render it to html
+
+    *Description of functions below:
+        *getData() asyncronously fetched data from github and calls functions to parse, filter and render component\
+        *SetHeader(repo) fetches and returns certain atributres from repo object
+        *SetRepos(repo) fetches and returns certain atributres from repo object
+        *injectHeader(header) dynamically renders the header to the body
+        *injectRepo(repo) creates a new repository elements and injects it into the body
+
+
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+async function getData (){
+    let response = await fetch('https://api.github.com/users/MohamedShatry/repos');
+    var resJSON = await response.json();
+    let varHeader = SetHeader(resJSON[0].owner);
+    injectHeader(varHeader);
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+    resJSON.forEach(repo => {
+        var returnRepo = SetRepos(repo);
+        injectRepo(returnRepo);
+    });
 }
+
+
+function SetHeader(repo){
+    var header = {
+        title : repo.login,
+        url: repo.html_url,
+        avatar_url: repo.avatar_url
+    }
+
+    return header;
+}
+
+function SetRepos(repo){
```

Comment:
This function is probably not needed (same for SetHeader) since it's only being called from one location. You can just implement it inline.

Reply:
+1 
inline to avoid jumping into another function.
getData() {
  owner := res[0].owner
  let varHeader = {
        title : repo.login,
        url: repo.html_url,
        avatar_url: repo.avatar_url
    }
}

Reply:
Resolved